### ISSUE TITLE
NNG fails in child process created using fork()

### ISSUE THREAD

**Issue by dilipatgit** [nanomsg/nng#1635]
Error:
**panic: nng is not fork-reentrant safe
This message is indicative of a BUG.
Report this at https://github.com/nanomsg/nng/issues**

**Comment by gdamore** [comment]
Yep. If you have a program that tries to fork() and then tries to use NNG in the child (without calling exec) then it fails.  There are a lot of problems with fork() and threads - too many for NNG to attempt to solve. (Fork and pthreads are basically inherently incompatible.)
